x-pack/plugins/infra/server/lib/infra_types.ts

```diff
@@ -32,5 +32,6 @@ export interface InfraBackendLibs extends InfraDomainLibs {
   handleEsError: typeof handleEsError;
   logsRules: RulesServiceSetup;
   metricsRules: RulesServiceSetup;
+  kibanaBaseUrl: string | undefined;
```

Comment:
Adding this plain and optional string reduces the ergonomics of the already suboptimal `libs` object further IMHO. How about we inject the core `http` or maybe the `IBasePath` here (https://github.com/elastic/kibana/blob/main/docs/development/core/server/kibana-plugin-core-server.basepath.md)? The consumers could then use the `prepend` method to avoid manual and error-prone url string manipulation.

Reply:
@weltenwort, The`IBasePath`seems really interesting, I didn't know about it!
I did a fast check, It seems that I only need to `import { IBasePath } from 'kibana/public'`  where I want to use it, then call the related functions. No more core plugin injection in the setup function of the Logs plugin. Is that right? If I got that correctly, that is excellent!!

Reply:
I think I see now how I can use `IBasePath`

Reply:
@weltenwort, thanks for the suggestion. I updated the code using `IBasePath`, and it did indeed simplify the workflow. 
But I still needed to use the `URL` and `path` API from Node unless I wanted to concatenate the string simply like `+` or string templating.
The `prepend` function from `IBasePath` will only concatenate the basePath and not the `publicBaseUrl` which I need to construct the full URL